x-pack/plugins/ingest_manager/server/services/agent_config.ts

```diff
@@ -336,7 +336,7 @@ class AgentConfigService {
       throw new Error('Agent configuration not found');
     }
 
-    const defaultConfigId = await this.getDefaultAgentConfigId(soClient);
+    const { id: defaultConfigId } = await this.ensureDefaultAgentConfig(soClient);
```

Comment:
Not sure it's the perfect solution, as it's going to eventually create the default config outside of the setup (not sure of the side effect of this)

What do you think of catching 404 of the previous call? 


something like 
```js
try {
  const defaultConfigId = await this.getDefaultAgentConfigId(soClient);

  if (id === defaultConfigId) {
      throw new Error('The default agent configuration cannot be deleted');	      throw new Error('The default agent configuration cannot be deleted');
    }	    }
} catch (err) {
  // implement real test here :) 
  if (err !== 404) { throw err; }
}